### DISCUSSION TITLE
What is the atomic file?

### DISCUSSION THREAD

**Discussion by mayaKaplansky** [RUCAIBox/RecBole#607] [Q&A] [answered]
Hi,
lets say I have my own dataset, and convert it to the tables as described. When I want to actually save them as atomic files, do I just change the file type manually?

**Comment by hyp1231** [comment, marked as answer]
Sure it is.

You need to pay attention to two parts mainly:
1. The first line, which contains multiple columns in ``field_name:field_type`` format.
2. The suffixes of files, as you mentioned, indicate the file type.

See [https://github.com/RUCAIBox/RecBole/tree/master/tests/test_data/test](https://github.com/RUCAIBox/RecBole/tree/master/tests/test_data/test) for examples of atomic file.
[https://github.com/RUCAIBox/RecBole/blob/511cda89deca404808bea1df2f5e98e17e9015f1/tests/test_data/test/test.inter#L1-L4](https://github.com/RUCAIBox/RecBole/blob/511cda89deca404808bea1df2f5e98e17e9015f1/tests/test_data/test/test.inter#L1-L4)

**Reply by mayaKaplansky** [reply]
Thanks! So I have to create a dataframe in this format, and then write it line by line into a new txt file.
Then change the extension from txt to inter?
I cannot use pickle dump?

**Reply by hyp1231** [reply]
> Thanks! So I have to create a dataframe in this format, and then write it line by line into a new txt file.
> Then change the extension from txt to inter?
> I cannot use pickle dump?

As long as the text file fit the format mentioned above, it's ok. So write a ``pandas.DataFrame`` line by line is one of the acceptable ways.

Pickle dump is not available yet.

**Reply by mayaKaplansky** [reply]
Thank you! In all atomic file types, the delimiter is a tab? I think I saw in some of the examples a space.

**Reply by hyp1231** [reply]
The delimiter in atomic files can be arbitrary, which should be the same with ``config['field_separator']``.
By default, ``config['field_separator'] == '\t'``.

Please refer to [Atomic File Format](https://recbole.io/docs/user_guide/data/data_args.html#atomic-file-format) for more details.

**Reply by mayaKaplansky** [reply]
Thank you!
